python/src/scipp/plot/slicer.py

```diff
@@ -216,3 +219,64 @@ def make_slider_label(self, var, indx):
 
     def mask_to_float(self, mask, var):
         return np.where(mask, var, None).astype(np.float)
+
+    def axis_to_dim_label(self, axis):
+        """
+        Get dimensions and label (if present) from requested axis
+        """
+        ticks = None
+        if isinstance(axis, Dim):
+            dim = axis
+            lab = None
+            make_fake_coord = False
+            fake_unit = None
+            if not self.data_array.coords.__contains__(dim):
+                make_fake_coord = True
+            else:
+                tp = self.data_array.coords[dim].dtype
+                if tp == dtype.vector_3_float64:
+                    make_fake_coord = True
+                    ticks = ["(" + ",".join([value_to_string(item, precision=2)
+                                             for item in elem]) + ")"
+                             for elem in self.data_array.coords[dim].values]
```

Comment:
Thinking of, e.g., DREAM with 10 million pixels, this conversion might be quite costly? In reality I suppose we are only using a tiny fraction of the labels, is there an easy way to avoid converting everything?

Reply:
Actually, there totally is a way to do this. Much better idea, thanks! Will do that today.